pkg/stanza/operator/input/windows/input.go

```diff
@@ -232,88 +233,72 @@ func (i *Input) read(ctx context.Context) int {
 
 // processEvent will process and send an event retrieved from windows event log.
 func (i *Input) processEvent(ctx context.Context, event Event) {
-	var providerName string // The provider name is only retrieved if needed.
-	if !i.raw || len(i.excludeProviders) > 0 {
-		var err error
-		providerName, err = event.GetPublisherName(i.buffer)
-		if err != nil {
-			i.Logger().Error("Failed to get provider name", zap.Error(err))
-			return
-		}
+	providerName, err := event.GetPublisherName(i.buffer)
```

Comment:
nit: getting the publisher name can be skipped under two conditions: 

1. `supressRenderingInfo` is set to true
2. there is no provider excluded. 

Hande-compiled it will break if taken as is:
```suggestion
	var providerName string // The provider name is only retrieved if needed.
	if !i.supressRenderingInfo || len(i.excludeProviders) > 0 {
		providerName, err := event.GetPublisherName(i.buffer)
```

The default value will work with the remaining code.

Reply:
I am not smart enough to keep track of all the considerations in this function so I refactored into smaller functions which I think separate the concerns better. Hopefully it is correct now as well.

Reply:
As part of this I found error handling to be awkward. We always log errors at error level, except in one case where it is a warning. I think they could all just be errors that are returned until the caller of `processEvent` logs them. [POC here](https://github.com/djaglowski/opentelemetry-collector-contrib/compare/wel-supress-rendering-info...djaglowski:opentelemetry-collector-contrib:wel-error-cleanup), but I suspect this is a follow up PR as this one is already too complicated.

Reply:
Breaking the code in functions made it much better.

Reply:
The POC also looks good, small nit: the name of the functions should be like `renderSimpleAndSend` and `renderDeepAndSend`.

Reply:
Thanks, I've pushed naming changes but I'll hold the error handling for another PR.